hugegraph-core/src/main/java/com/baidu/hugegraph/backend/serializer/BytesBuffer.java

```diff
@@ -332,14 +333,27 @@ public String readString() {
         return StringEncoding.decode(this.readBytes());
     }
 
-    public BytesBuffer writeStringWithEnding(String val) {
-        if (!val.isEmpty()) {
-            byte[] bytes = StringEncoding.encode(val);
-            // assert '0xff' not exist in string-id-with-ending (utf8 bytes)
-            assert !Bytes.contains(bytes, STRING_ENDING_BYTE);
+    public BytesBuffer writeStringWithEnding(String value) {
+        if (!value.isEmpty()) {
+            byte[] bytes = StringEncoding.encode(value);
+            /*
+             * assert '0x00'/'0xFF' not exist in string index id
+             * NOTE:
+             *   0x00 is NULL in UTF8(or ASCII) bytes
+             *   0xFF is not a valid byte in UTF8 bytes
+             */
+            assert !Bytes.contains(bytes, STRING_ENDING_BYTE_FF) :
```

Comment:
if not use 0xff at all, prefer delete it and no need to assert

Reply:
to preserve the byte for future usages